--- C:/wxWidgets-2.8.7/include/wx/aui/auibook.h	Fri Nov 09 14:16:10 2007
+++ E:/Dev/wxAUI/auibook.h	Thu May 29 12:50:40 2008
@@ -274,6 +274,9 @@
     wxBitmap m_active_windowlist_bmp;
     wxBitmap m_disabled_windowlist_bmp;
 
+	wxBitmap m_pressed_close_bmp;
+	wxBitmap m_hot_close_bmp;
+
     int m_fixed_tab_width;
     int m_tab_ctrl_height;
     unsigned int m_flags;
@@ -394,7 +397,7 @@
     void SetNoneActive();
     int GetActivePage() const;
     bool TabHitTest(int x, int y, wxWindow** hit) const;
-    bool ButtonHitTest(int x, int y, wxAuiTabContainerButton** hit) const;
+    bool ButtonHitTest(int x, int y, wxAuiTabContainerButton** hit, bool ignoreDisabled=true) const;
     wxWindow* GetWindowFromIdx(size_t idx) const;
     int GetIdxFromWindow(wxWindow* page) const;
     size_t GetPageCount() const;
@@ -581,6 +584,12 @@
     // Shows the window menu
     bool ShowWindowMenu();
 #endif
+
+    int PageToTab(size_t page_idx);
+    int TabToPage(size_t tab_idx);
+
+	wxString SavePerspective();
+	bool LoadPerspective(const wxString& layout);
 
 protected:
 
