diff -Naurp WebKit-r43110/JavaScriptCore/jscore.bkl webkit/JavaScriptCore/jscore.bkl
--- WebKit-r43110/JavaScriptCore/jscore.bkl	2009-04-30 23:01:27.000000000 +0400
+++ webkit/JavaScriptCore/jscore.bkl	2009-05-02 12:40:13.000000000 +0400
@@ -105,40 +105,6 @@ JavaScriptCore Bakefile project file.
 
     </template>
     
-    <exe id="jsc" template="icu,jscore,pthreads,wxwk">
-        <cxx-rtti>off</cxx-rtti>
-        <cxx-exceptions>off</cxx-exceptions>
-        <debug-info>on</debug-info>
-        <depends>jscore</depends>
-        <include>$(SRCDIR)</include>
-        <include>$(WK_ROOT)/JavaScriptCore</include>
-        <include>$(WK_ROOT)/JavaScriptCore/assembler</include>
-        <include>$(WK_ROOT)/JavaScriptCore/bytecompiler</include>
-        <include>$(WK_ROOT)/JavaScriptCore/debugger</include>
-        <include>$(WK_ROOT)/JavaScriptCore/parser</include>
-        <include>$(WK_ROOT)/JavaScriptCore/pcre</include>
-        <include>$(WK_ROOT)/JavaScriptCore/profiler</include>
-        <include>$(WK_ROOT)/JavaScriptCore/runtime</include>
-        <include>$(WK_ROOT)/JavaScriptCore/interpreter</include>
-        <include>$(WK_ROOT)/JavaScriptCore/bytecode</include>
-        <include>$(WK_ROOT)/JavaScriptCore/jit</include>
-        <include>$(WK_ROOT)/JavaScriptCore/wrec</include>
-        <include>$(WK_ROOT)/JavaScriptCore/wtf</include>
-        <dirname>$(WKOUTPUTDIR)</dirname>
-        <sources>$(SRCDIR)/jsc.cpp</sources>
-        <set var="READLINE_LIB">
-            <if cond="WX_PORT=='mac'">edit</if>
-        </set>
-        <sys-lib>$(READLINE_LIB)</sys-lib>
-        <if cond="FORMAT in ['msvc','msvs2005prj']">
-            <include>$(WK_ROOT)/WebKitLibraries/win/include</include>
-            <sys-lib>winmm</sys-lib> <!-- for timeGetTime -->
-            <lib-path>$(WKOUTPUTDIR)</lib-path>
-            <lib-path>$(WK_ROOT)/WebKitLibraries/win/lib</lib-path>
-        </if>
-        
-    </exe>
-
     <action id="DerivedSources">
         <is-phony />
         <command>bash make-generated-sources.sh</command>
diff -Naurp WebKit-r43110/WebKit/wx/Bakefiles.bkgen webkit/WebKit/wx/Bakefiles.bkgen
--- WebKit-r43110/WebKit/wx/Bakefiles.bkgen	2009-04-30 04:04:21.000000000 +0400
+++ webkit/WebKit/wx/Bakefiles.bkgen	2009-05-02 12:38:57.000000000 +0400
@@ -41,19 +41,13 @@ Build configuration script for Bakefile.
 
     <!-- List of output formats to generate: -->
     <add-formats>
-        gnu,msvs2005prj,msvc
+        gnu
     </add-formats>
 
     <!-- JSCore settings -->
     <add-flags files="../../JavaScriptCore/jscore.bkl" formats="gnu">
         -o../../JavaScriptCore/GNUmakefile
     </add-flags>
-    <add-flags files="../../JavaScriptCore/jscore.bkl" formats="msvs2005prj">
-        -o../../JavaScriptCore/JavaScriptCore.sln
-    </add-flags>
-    <add-flags files="../../JavaScriptCore/jscore.bkl" formats="msvc">
-        -o../../JavaScriptCore/makefile.vc
-    </add-flags>
     <!-- testing XCode backend
     <add-flags files="jscore.bkl" formats="xcode2">
         -o../WebCore/Projects/base/WebCoreBase.xcode
@@ -64,42 +58,18 @@ Build configuration script for Bakefile.
     <add-flags files="../../WebCore/webcore-wx.bkl" formats="gnu">
         -o../../WebCore/GNUmakefile
     </add-flags>
-    <add-flags files="../../WebCore/webcore-wx.bkl" formats="msvs2005prj">
-        -o../../WebCore/webcore-wx.sln
-    </add-flags>
-    <add-flags files="../../WebCore/webcore-wx.bkl" formats="msvc">
-        -o../../WebCore/makefile.vc
-    </add-flags>
     
     <add-flags files="wxwebkit.bkl" formats="gnu">
         -oGNUmakefile
     </add-flags>
-    <add-flags files="wxwebkit.bkl" formats="msvs2005prj">
-        -owxWebKit.sln
-    </add-flags>
-    <add-flags files="wxwebkit.bkl" formats="msvc">
-        -omakefile.vc
-    </add-flags>
     
     <add-flags files="bindings/python/wxwebkit-py.bkl" formats="gnu">
         -obindings/python/GNUmakefile
     </add-flags>
-    <add-flags files="bindings/python/wxwebkit-py.bkl" formats="msvs2005prj">
-        -obindings/python/wxWebKit.sln
-    </add-flags>
-    <add-flags files="bindings/python/wxwebkit-py.bkl" formats="msvc">
-        -obindings/python/makefile.vc
-    </add-flags>
     
     <add-flags files="../../WebKitTools/wx/browser/browser.bkl" formats="gnu">
         -o../../WebKitTools/wx/browser/GNUmakefile
     </add-flags>
-    <add-flags files="../../WebKitTools/wx/browser/browser.bkl" formats="msvs2005prj">
-        -o../../WebKitTools/wx/browser/browser.sln
-    </add-flags>
-    <add-flags files="../../WebKitTools/wx/browser/browser.bkl" formats="msvc">
-        -o../../WebKitTools/wx/browser/makefile.vc
-    </add-flags>
     <!-- testing XCode backend 
     <add-flags files="webcore-wx.bkl" formats="xcode2">
         -o../WebCore/Projects/wx/wxWebCore.xcode
diff -Naurp WebKit-r43110/WebKitTools/wx/build-wxwebkit webkit/WebKitTools/wx/build-wxwebkit
--- WebKit-r43110/WebKitTools/wx/build-wxwebkit	2009-05-01 08:15:10.000000000 +0400
+++ webkit/WebKitTools/wx/build-wxwebkit	2009-05-02 12:38:57.000000000 +0400
@@ -302,7 +302,6 @@ if [ $do_build == 1 -o $do_clean == 1 ];
 
     do_make $WK_ROOT/WebCore "-DBUILDING_WX__=1 $WXGC_DEFINE $EXTRA_CPPFLAGS" WX_DEBUG=$wxdebug WEBKIT_ROOT=$WK_ROOTDIR DEBUG=$debug PLATFORM_OS=$PLATFORM_OS $other_args
     do_make $WK_ROOT/WebKit/wx "-DBUILDING_WX__=1 -DWXMAKINGDLL_WEBKIT=1 $WXGC_DEFINE $EXTRA_CPPFLAGS" WX_DEBUG=$wxdebug DEBUG=$debug WEBKIT_ROOT=$WK_ROOTDIR PLATFORM_OS=$PLATFORM_OS $other_args 
-    do_make $WK_ROOT/WebKitTools/wx/browser  "-DBUILDING_WX__=1 -DWXUSINGDLL_WEBKIT=1 $WXGC_DEFINE $EXTRA_CPPFLAGS" WX_DEBUG=$wxdebug DEBUG=$debug WEBKIT_ROOT=$WK_ROOTDIR PLATFORM_OS=$PLATFORM_OS $other_args 
 
     if [ $do_clean == 1 ]; then
         rm -rf $WK_ROOT/JavaScriptCore/DerivedSources
