From bug 24345


Index: WebKit/wx/WebFrame.h
===================================================================
--- WebKit/wx/WebFrame.h	(revision 41418)
+++ WebKit/wx/WebFrame.h	(working copy)
@@ -46,6 +46,9 @@
 
 #ifndef SWIG
 
+#if !wxCHECK_VERSION(2,9,0) && wxCHECK_GCC_VERSION(4,0)
+#define WXDLLIMPEXP_WEBKIT __attribute__ ((visibility("default")))
+#else
 #if WXMAKINGDLL_WEBKIT
 #define WXDLLIMPEXP_WEBKIT WXEXPORT
 #elif defined(WXUSINGDLL_WEBKIT)
@@ -54,8 +57,7 @@
 #define WXDLLIMPEXP_WEBKIT
 #endif
 
-#else 
-#define WXDLLIMPEXP_WEBKIT
+#endif
 #endif // SWIG
 
 class WXDLLIMPEXP_WEBKIT wxWebViewDOMElementInfo
Index: WebKit/wx/WebView.h
===================================================================
--- WebKit/wx/WebView.h	(revision 41418)
+++ WebKit/wx/WebView.h	(working copy)
@@ -46,6 +46,9 @@
 
 #ifndef SWIG
 
+#if !wxCHECK_VERSION(2,9,0) && wxCHECK_GCC_VERSION(4,0)
+#define WXDLLIMPEXP_WEBKIT __attribute__ ((visibility("default")))
+#else
 #if WXMAKINGDLL_WEBKIT
 #define WXDLLIMPEXP_WEBKIT WXEXPORT
 #elif defined(WXUSINGDLL_WEBKIT)
@@ -54,8 +57,7 @@
 #define WXDLLIMPEXP_WEBKIT
 #endif
 
-#else 
-#define WXDLLIMPEXP_WEBKIT
+#endif
 #endif // SWIG
 
 #ifndef SWIG
Index: WebKitTools/wx/build-wxwebkit
===================================================================
--- WebKitTools/wx/build-wxwebkit	(revision 41418)
+++ WebKitTools/wx/build-wxwebkit	(working copy)
@@ -179,7 +179,7 @@
         cd $olddir
     else
         if [ -f $dir/GNUmakefile ]; then
-            make -C $dir -f GNUmakefile $MAKE_ARGS CXXFLAGS="$cxxflags" $@
+            make -C $dir -f GNUmakefile $MAKE_ARGS CXXFLAGS="$cxxflags -fvisibility=hidden -fvisibility-inlines-hidden" $@
         fi
     fi
     if [ $? != 0 ]; then
