--- WebKit/WebCore/platform/wx/SharedTimerWx.cpp	2009-05-05 03:17:51.000000000 +0400
+++ webkit/WebCore/platform/wx/SharedTimerWx.cpp	2009-05-05 03:19:07.000000000 +0400
@@ -77,11 +77,7 @@ void setSharedTimerFireTime(double fireT
     if (!wkTimer)
         wkTimer = new WebKitTimer();
         
-    unsigned int intervalInMS = interval * 1000;
-
-    // sanity check
-    if (intervalInMS < 1)
-        intervalInMS = 1;
+    unsigned int intervalInMS = interval < 0.0012 ? 1 : interval * 1000;
 
     wkTimer->Start(intervalInMS, wxTIMER_ONE_SHOT);
 }
